.github/workflows/pull_request.yml

```diff
@@ -22,13 +22,23 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: ${{ env.PYTHON_VERSION }}
-          cache: 'pipenv'
-      - name: Install Pipenv
-        run: pip install pipenv==2023.8.22
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.7.1
+          virtualenvs-create: true
+          virtualenvs-in-project: false
+          virtualenvs-path: ~/.virtualenvs
+      - name: Cache Poetry virtualenv
+        uses: actions/cache@v3
+        id: cache-virtualenv
+        with:
+          path: ~/.virtualenvs
+          key: ${{ runner.os }}-${{ env.PYTHON_VERSION }}-virtualenvs-${{ hashFiles('poetry.lock') }}
```

Comment:
We should cache using the `cache` key of `actions/setup-python@v4` step.  Poetry is supported. https://github.com/actions/setup-python#caching-packages-dependencies

Reply:
Forgot to update. Was added 👍